scripts/src/metrics/metrics.py

```diff
@@ -55,35 +59,301 @@ def send_release_metrics(write_key, downloads):
+                    pr_content = "report only"
+            elif tgz_found:
+                pr_content = "tgz only"
+            elif src_found:
+                pr_content = "src only"
+
+            return pr_content,type,org,chart,version
+
+    return pr_content,"","","",""
+
+def check_and_get_pr_content(pr,repo):
+    repo_name = repo.full_name
+    if (pr.user.login in ignore_users and pr.user.login not in repo_name) or pr.draft or pr.base.ref != "main":
+        print(f"[INFO] Ignore pr, user: {pr.user.login}, draft: {pr.draft}, target_branch: {pr.base.ref}")
+        return "not-chart","","","",""
+
+    return get_pr_content(pr)
+    
+
+def process_pr(write_key,repo,message_file,pr_number,action):
+    pr = repo.get_pull(int(pr_number))
+
+    pr_content,type,provider,chart,version = check_and_get_pr_content(pr,repo)
+    if pr_content != "not-chart":
+        if action == "opened":
+            send_submission_metric(write_key,type,provider,chart,pr_number,pr_content)
+
+        pr_result = process_comment_file(message_file,pr_number)
+        num_fails=0
+        if pr_result == "report-failure":
+            num_fails,checks_failed = process_report_fails(message_file)
+            for check in checks_failed:
+                send_check_metric(write_key,type,provider,chart,pr_number,check)
+        elif pr_result == "content-failure":
+            num_fails = 1
+        send_outcome_metric(write_key,type,provider,chart,pr_number,pr_result,num_fails)
+
+        ## if pr is merged we can collect summary stats
+        if pr.merged_at:
+
+            builds =  process_comments(repo,pr)
+            print(f"[INFO]    PR  build cycles : {builds}")
+            builds_out = str(builds)
+            if builds > 5:
+                builds_out = "> 5"
+
+            elapsed_time = pr.merged_at - pr.created_at
+            # round up to an hour to avoid 0 time
+            elapsed_hours = elapsed_time.total_seconds()//3600
+            duration = "0-1 hours"
+            if 24 > elapsed_hours > 1:
+                duration = "1-24 hours"
+            elif 168 > elapsed_hours > 24:
+                duration = "1-7 days"
+            elif elapsed_hours > 168:
+                duration= "> 7 days"
+
+            send_merge_metric(write_key,type,provider,chart,duration,pr_number,builds_out,pr_content)
+
+
+def send_summary_metric(write_key,num_submissions,num_merged,num_abandoned,num_in_progress,num_partners,num_charts):
+    properties = { "submissions": num_submissions, "merged": num_merged, "abandoned" : num_abandoned, "in_progress" : num_in_progress,
+                   "partners": num_partners, "partner_charts" : num_charts}
+    id = "helm-metric-summary"
+
+    send_metric(write_key,id,"PR Summary",properties)
+
+def send_outcome_metric(write_key,type,provider,chart,pr_number,outcome,num_fails):
+
+    properties = { "type": type, "provider": provider, "chart" : chart, "pr" : pr_number, "outcome" : outcome, "failures" :  num_fails}
```

Comment:
Is the value of "provider" is correct here?

Reply:
Also I invited you to woopra. Once in you should see the data here:
https://app.woopra.com/project/test.openshift.helm.charts/activity/activity:default